src/Model/Metadata.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\EuropaSearchClient\Model;
+
+/**
+ * A class that represents a document's metadata.
+ */
+class Metadata
+{
+
+    protected $key;
+
+    protected $value;
+
+    /**
+     * Returns the key name.
+     *
+     * @return string
+     *   The metadata key.
+     */
+    public function getKey(): string
+    {
+        return $this->key;
+    }
+
+    /**
+     * Sets the object key.
+     *
+     * @param string $key
+     *   The object key name.
+     *
+     * @return $this
+     */
+    public function setKey(string $key): Metadata
```

Comment:
lets say self since we dont have an interface for this, same for the others in this file.
```suggestion
    public function setKey(string $key): self
```